test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart

```diff
@@ -0,0 +1,455 @@
+          return Dom.ul()(
+            Dom.li()('Foo: ', props.foo),
+            Dom.li()('Bar: ', props.bar),
+          );
+        }
+      }
+    ''',
+            expectedOutput: r'''
+      @Factory()
+      UiFactory<FooProps> Foo =
+          // ignore: undefined_identifier
+          $Foo;
+
+      @Props()
+      mixin FooPropsMixin on UiProps {
+        String foo;
+        int bar;
+      }
+      
+      class FooProps = UiProps with ADifferentPropsClassMixin, FooPropsMixin;
+
+      @Component2()
+      class FooComponent extends UiComponent2<FooProps> {
+        @override
+        render() {
+          return Dom.ul()(
+            Dom.li()('Foo: ', props.foo),
+            Dom.li()('Bar: ', props.bar),
+          );
+        }
+      }
+    ''',
+          );
+
+          expect(propsAndStateClassNamesConvertedToNewBoilerplate, {
+            'FooProps': 'FooPropsMixin',
+          });
+        });
+
+        test('and the class uses mixins', () {
+          testSuggestor(
+            expectedPatchCount: 6,
+            input: r'''
+      @Factory()
+      UiFactory<FooProps> Foo =
+          // ignore: undefined_identifier
+          $Foo;
+
+      @Props()
+      class _$FooProps extends UiProps with AMixin, AnotherMixin {
+        String foo;
+        int bar;
+      }
+
+      @Component2()
+      class FooComponent extends UiComponent2<FooProps> {
+        @override
+        render() {
+          return Dom.ul()(
+            Dom.li()('Foo: ', props.foo),
+            Dom.li()('Bar: ', props.bar),
+          );
+        }
+      }
+    ''',
+            expectedOutput: r'''
+      @Factory()
+      UiFactory<FooProps> Foo =
+          // ignore: undefined_identifier
+          $Foo;
+
+      @Props()
+      mixin FooPropsMixin on UiProps {
+        String foo;
+        int bar;
+      }
+      
+      class FooProps = UiProps with FooPropsMixin, AMixin, AnotherMixin;
+
+      @Component2()
+      class FooComponent extends UiComponent2<FooProps> {
+        @override
+        render() {
+          return Dom.ul()(
+            Dom.li()('Foo: ', props.foo),
+            Dom.li()('Bar: ', props.bar),
+          );
+        }
+      }
+    ''',
+          );
+
+          expect(propsAndStateClassNamesConvertedToNewBoilerplate, {
+            'FooProps': 'FooPropsMixin',
+          });
+        });
+      });
+
+      test(
+          'and have a custom class with mixins when `shouldMigrateCustomClassAndMixins` is true',
+          () {
+        final testSuggestorWithFlag = getSuggestorTester(
+            AdvancedPropsAndStateClassMigrator(
+                shouldMigrateCustomClassAndMixins: true));
+
+        testSuggestorWithFlag(
+          expectedPatchCount: 14,
+          input: r'''
+        @Factory()
+        UiFactory<FooProps> Foo =
+            // ignore: undefined_identifier
+            $Foo;
+
+        @Props()
+        class _$FooProps extends ADifferentPropsClass with AMixin, AnotherMixin {
```

Comment:
What about the cases where:

* `ADifferentPropsClass` is the one that has mixins
* The mixins either found in this `with` clause, or on the `with` clause of `ADifferentPropsClass` implement more than just `UiProps`? 
    * _Those interfaces would need to get mixed into the final concrete props class as well._

Reply:
Offline resolution - always add a `// FIXME` comment when a class extends from a class that is not `UiProps`.